frontend/src/components/chatInput/ChatInput.tsx

```diff
@@ -1,30 +1,42 @@
 import {
-	Form,
-	InputWrapper,
-	MemeButton,
-	MemeGenerateIcon,
-	MessageInput
+    Form,
+    InputWrapper,
+    MemeButton,
+    MemeGenerateIcon,
+    MessageInput
 } from "./ChatInput.styled";
 import useChatInput from "./useChatInput";
+import GifWidget from "components/GifWidget/GifWidget";
+import { useState } from "react";
 
 const ChatInput = () => {
-	const { currentInputValue, handleSetCurrentInputValue, handleSubmit } =
-		useChatInput();
+    const { currentInputValue, handleSetCurrentInputValue, handleSubmit } =
+        useChatInput();
+    const [gifWidgetVisible, setGifWidgetVisible] = useState(false);
```

Comment:
this state must be inside "useChatInput" cause u do this hook, because we want to save our logic part of components here.